delphi-checks/src/test/java/au/com/integradev/delphi/checks/IterationPastHighBoundCheckTest.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * Sonar Delphi Plugin
+ * Copyright (C) 2025 Integrated Application Development
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package au.com.integradev.delphi.checks;
+
+import au.com.integradev.delphi.builders.DelphiTestUnitBuilder;
+import au.com.integradev.delphi.checks.verifier.CheckVerifier;
+import org.junit.jupiter.api.Test;
+
+class IterationPastHighBoundCheckTest {
+  @Test
+  void testIterationToLengthMinusOneShouldNotAddIssue() {
+    CheckVerifier.newVerifier()
+        .withCheck(new IterationPastHighBoundCheck())
+        .onFile(
+            new DelphiTestUnitBuilder()
+                .appendImpl("procedure Test(MyArr: TArray<string>);")
+                .appendImpl("var")
+                .appendImpl("  I: Integer;")
+                .appendImpl("begin")
+                .appendImpl("  for I := 0 to Length(MyArr) - 1 do begin")
```

Comment:
Can we also check if using `Pred()` [https://docwiki.embarcadero.com/Libraries/Athens/en/System.Pred] instead of `-1` works?
`.appendImpl(" for I := 0 to Pred(Length(MyArr)) do begin")`

Reply:
Because of the way the rule works, this test isn't asserting that the `-1` is detected as such - it's detecting that the expression is not of the form `Length(...)`. So something like `Pred(Length(...))` is already correctly ignored.

That being said, I'll add the test - could be a useful addition. Thanks!

Reply:
Ok, thanks for the clarification